src/react/DeviceStatusPage/AliasesCard.tsx

```diff
@@ -0,0 +1,106 @@
+/*
+   This file is part of Astarte.
+
+   Copyright 2020 Ispirata Srl
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+import React from 'react';
+import { Button, Card, Table } from 'react-bootstrap';
+
+import type { AstarteDevice } from 'astarte-client';
+import FullHeightCard from '../components/FullHeightCard';
+
+interface AliasKeyValuePair {
+  key: string;
+  value: string
+}
+
+interface AliasesTableProps {
+  aliases: Map<string, string>;
+  onEditAliasClick: (key: string) => void;
+  onRemoveAliasClick: ({key, value}: AliasKeyValuePair) => void;
```

Comment:
I saw you introduced a file for each Modal of the page. However in each file you are basically passing props to an already defined component. Another approach could be to just inline the modals since you are not reusing them multiple times.
For example, inline DeleteAliasModal, AddAliasModal and EditAliasModal in this AliasesCard. Then you could just notify the parent Page on successful confirmations of modals:
```
onUpdateAlias (key: string, newValue: AliasKeyValuePair) => void;
onRemoveAlias: (key: string) => void;
```
So the parent page has fewer state to keep track of: just whether an alias is being removed or updated, but not which alias.
The same reasoning goes for the other Cards / Modals.
Would you think it could be feasible?

Reply:
there are 2 problems.
1) the confirm action from the modal requires communication with Astarte, and then the modal must be notified (show spinner).
2) we want to make sure it's impossible to have 2 modals opened at the same time

Those 2 together forces the state to be handled in the parent component